### PR TITLE
react_pagination solution

### PR DESCRIPTION
**Pull request by vladyslav-harkusha**
[DEMO LINK](https://vladyslav-harkusha.github.io/react_pagination/)

### CODE REVIEW

**Review by illia-sosnytskyi — APPROVED** [review]
nice

File: src/App.tsx [hunk 1]
```diff
@@ -1,18 +1,39 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (sergii-nosachenko) [inline src/App.tsx:6]: Cleanup code

File: src/App.tsx [hunk 2]
```diff
@@ -1,18 +1,39 @@
...
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+const items = getNumbers(1, 42).map(n => `Item ${n}`);
```
Comment (sergii-nosachenko) [inline src/App.tsx:7]: Create constant for 42 and use it everywhere

File: src/App.tsx [hunk 3]
```diff
@@ -1,18 +1,39 @@
...
+  };
+
+  const onPageChange = (page: number) => {
+    setCurrentPage(page);
+  };
+
+  const startItemIndex = currentPage * itemsPerPage - itemsPerPage;
+  const endItemIndex = currentPage * itemsPerPage <= 42
+    ? currentPage * itemsPerPage
+    : 42;
```
Comment (sergii-nosachenko) [inline src/App.tsx:27]: Items count can be !== 42. Your code must work for any number

File: src/App.tsx [hunk 4]
```diff
@@ -34,62 +57,16 @@ export const App: React.FC = () => {
...
-          >
-            »
-          </a>
-        </li>
-      </ul>
+      <Pagination
+        totalPages={totalPages}
+        currentPage={currentPage}
+        onPageChange={onPageChange}
+      />
```
Comment (sergii-nosachenko) [inline src/App.tsx:64]: This component accepts another props. Check readme first and fix

File: src/components/Pagination/Pagination.tsx [hunk 5]
```diff
@@ -1 +1,61 @@
...
+}) => {
+  return (
+    <ul className="pagination">
+      <li className={cn('page-item', { disabled: currentPage === 1 })}>
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled={currentPage === 1}
+          onClick={() => currentPage !== 1 && onPageChange((currentPage - 1))}
```
Comment (sergii-nosachenko) [inline src/components/Pagination/Pagination.tsx:23]: Create handler

File: src/components/Pagination/Pagination.tsx [hunk 6]
```diff
@@ -1 +1,61 @@
...
+      <li
+        className={cn('page-item', { disabled: currentPage === totalPages })}
+      >
+        <a
+          data-cy="nextLink"
+          className="page-link"
+          href="#next"
+          aria-disabled={currentPage === totalPages}
+          onClick={() => currentPage !== totalPages
+            && onPageChange((currentPage + 1))}
```
Comment (sergii-nosachenko) [inline src/components/Pagination/Pagination.tsx:54]: Create handler

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 7]
```diff
@@ -1,18 +1,37 @@
...
+
+  const visibleItems = items.slice(startItemIndex, endItemIndex);
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${startItemIndex + 1} - ${endItemIndex} of 42)`}
```
Comment (sergii-nosachenko) [inline src/App.tsx:34]: It works only for 42 items.

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]
One small fix needed to go

**Review by sergii-nosachenko — APPROVED** [review]
Good job 👍👍👍

**Pull request closed without merging**